### PR TITLE
Authentication routes on dev 

### PR DESCRIPTION
**Pull request by sanya-mahajan**
#34 
made the register and login routes using token

### CODE REVIEW

**Comment by codervivek5** [comment]
@sanya-mahajan can I see your changes so that I can review

**Comment by sanya-mahajan** [comment]
Screenshot from 2023-05-26 23-30-24
Screenshot from 2023-05-26 23-30-14

File: Backend Code/VigyBag/users/views.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        else:
+            return Response(serializer.errors,status=status.HTTP_400_BAD_REQUEST)
+
+#allow posting to login view to get token
+
+class LoginView(APIView):
+    def post(self,request):
+        serializer = LoginSerializer(data=request.data)
+        if serializer.is_valid():
+            print(1)
```
Comment (JigarJoshi04) [inline Backend Code/VigyBag/users/views.py]: I think this is a redundant print and can be removed

File: Backend Code/VigyBag/users/views.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+from .models import *
+from .serializers import *
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.authentication import TokenAuthentication
+
+class RegisterView(APIView):
+    def post(self, request):
+        email=request.data['email']
+        if CustomUser.objects.filter(email=email).exists():
+            return Response({'status':'failed','message':'Email already exists'},status=status.HTTP_400_BAD_REQUEST)
```
Comment (JigarJoshi04) [inline Backend Code/VigyBag/users/views.py:21]: I think this check should happen after the serialising step

File: Backend Code/VigyBag/users/serializers.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    password=serializers.CharField(max_length=128,write_only=True)
+
+    class Meta:
+        model = CustomUser
+        fields = ('email', 'password')  
+        #have to add usertype ,default customer
+    
+    def create(self, validated_data):
+        user = CustomUser.objects.create_user(email=validated_data['email'], password=validated_data['password'])
+        return user
```
Comment (JigarJoshi04) [inline Backend Code/VigyBag/users/serializers.py]: This is a Create operation (An operation that should ideally be performed by repository of the code). It should not be placed in the serializer file. 

Serializers allow complex data such as querysets and model instances to be converted to native Python datatypes that can then be easily rendered into JSON, XML or other content types. Serializers also provide deserialization, allowing parsed data to be converted back into complex types, after first validating the incoming data.

Ref: https://www.django-rest-framework.org/api-guide/serializers/

File: Backend Code/VigyBag/users/serializers.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    class Meta:
+        model = CustomUser
+        fields = ('email', 'password')
+
+    def validate(self, data):
+        email=data.get('email', None)
+        password=data.get('password', None)
+        if email and password:
+            user=authenticate(email=email, password=password)
+            print(1234)
```
Comment (JigarJoshi04) [inline Backend Code/VigyBag/users/serializers.py]: Redundant print statement that can be removed.

File: Backend Code/VigyBag/users/serializers.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    password=serializers.CharField(max_length=128,write_only=True)
+    class Meta:
+        model = CustomUser
+        fields = ('email', 'password')
+
+    def validate(self, data):
+        email=data.get('email', None)
+        password=data.get('password', None)
+        if email and password:
+            user=authenticate(email=email, password=password)
```
Comment (JigarJoshi04) [inline Backend Code/VigyBag/users/serializers.py]: This is again a call to function of the buisness logic layer that serializer should not be making.
Suggesting to move business logic and triggers to business logic out of serializers.

**Review by JigarJoshi04 — CHANGES_REQUESTED** [review]

**Comment by sanya-mahajan** [comment]
Thank you for the feedback, would be working on it

**Comment by sanya-mahajan** [comment]
@JigarJoshi04 sir, kindly review the updates

**Review by codervivek5 — APPROVED** [review]

**Pull request merged**